Separate settings files for production, development and testing.

sgarciahelguera opened the pull request:
Mainly remove auth alternatives for production and testing, and add postgresql configuration for production.

Close #5

sgarciahelguera committed:
Separate settings files for production, development and testing.
Mainly remove auth alternatives for production and testing, and add
postgresql configuration for production.

Natiman58 commented on cubeseed/settings_test.py:
```diff
@@ -0,0 +1,20 @@
+
+REST_FRAMEWORK["TEST_REQUEST_DEFAULT_FORMAT"] = "json"
+REST_FRAMEWORK["TEST_REQUEST_RENDERER_CLASSES"] = [
+    "rest_framework.renderers.JSONRenderer",
```
how about handling file uploads of multipart files during testing ; may be can add Multipart renderer
"rest_framework.renderers.MultiPartRenderer" into our framework

sgarciahelguera replied:
Just merged from main and it looks like you added that alongside some test, right? 
It is now on test and development settings.

sgarciahelguera committed:
Merge branch 'main' into issue#5

sgarciahelguera committed:
Complete merge for AWS file storage related changes (including
extending the use of environ package to all settings that use environment)

sgarciahelguera committed:
better default values

sgarciahelguera committed:
I think this forces http for queries but I'm not sure. I am getting
inconsistent results were it fails sometimes and looks like it is
attempting https and getting rejected, so if it doesn't happen any
more I'll assume this works.

sgarciahelguera committed:
Update docker compose and dockerfile.local

sgarciahelguera committed:
Update dockerfile, attempt at pushing two dockers

sgarciahelguera committed:
Remove failing image

sgarciahelguera committed:
Merge remote-tracking branch 'origin/main' into issue#5

sgarciahelguera committed:
Add dev version back

sgarciahelguera committed:
add gunicorn

sgarciahelguera committed:
sort requirements

sgarciahelguera committed:
Try with newer versions

sgarciahelguera committed:
Fix tag

sgarciahelguera committed:
Add settings file to env, add -dev to development version VERSION

sgarciahelguera committed:
Update docker compose, fix db host for porduction

sgarciahelguera committed:
Add docker compose for dev & for github images

mihrab34 commented on docker/docker-compose-dev.yml:
```diff
@@ -0,0 +1,39 @@
+      - DB_USER=${DB_USER}
+      - DB_PASS=${DB_PASS}
+      - DB_HOST=db
+    command: manage.py runserver 0.0.0.0:8000
```
can you add a `.env.example` file so contributors can know what to include in their env files when running the docker container

mihrab34 replied:
Testing the auth token endpoint on localhost also gave cors error You might want to look into that

![cors error](https://github.com/Cubeseed/Backend/assets/35562131/b6a3daa1-3ddc-471f-8284-78752638b1a7)

sgarciahelguera replied:
Added `docker/.env` with the basics (enough for development).

Reid-T-W commented:
Looks great, just added one minor comment

sgarciahelguera committed:
Got rid of Dockerfile.local

sgarciahelguera committed:
Workaround swagger issue on FarmPlannerViewSet

sgarciahelguera committed:
Use gunicorn

sgarciahelguera committed:
Fix runner with script

sgarciahelguera committed:
Add 127.0.0.1 to allowed hosts in development

sgarciahelguera commented on cubeseed/settings.py:
added 127.0.0.1 to allowed hosts

sgarciahelguera committed:
Fix runner.

sgarciahelguera committed:
Move docker compose to the project root & fix some details

sgarciahelguera committed:
Merge remote-tracking branch 'origin/main' into issue#5

sgarciahelguera commented:
Ended up moving `docker/docker-compose-dev.yml` to `docker-compose.yml` as having a symbolic link was not trivial (context has to point to the root of the project, relatively, so if you run from docker folder is one thing but if you run from a symlink in some other directory it has to be something else).

sgarciahelguera commented:
Everything should be working now for development, I'll leave some work on production and probably removing the testing settings for another pull request.
Please review/approve with that in mind.

Reid-T-W commented:
When accessing the swagger endpoint it loads up but there is still some issue with the FarmPlanner, kindly take a look at it below.

https://docs.google.com/document/d/1T5uePXzeNDF4L3-_YAwyO2KQgcUKOrybr5ttwtaSuwo/edit?usp=sharing

The exact steps I followed are

1. Clone the issue#5 branch 
2. Added .env file
3. Run `docker-compose up`
4. Access the "http://localhost:8000/swagger/" enpoint

Reid-T-W commented:
When accessing the (GET) /userauth/groups/ endpoint i get the following error, kindly take a look via this link

https://docs.google.com/document/d/1WjOPYxjvWwAC4TicvSGyfkg62Hy9F1oQos3Q4PGftn4/edit?usp=sharing

The exact steps I followed are

Clone the issue#5 branch
Added .env file
Run docker-compose up
Access the (GET) /userauth/groups/ endpoint via swagger

Reid-T-W commented:
I also noticed that there is a workaround put in place to handle the FarmPlanner and Swagger issue, the swagger loads up but I'm still facing errors as shown in this link

https://docs.google.com/document/d/1T5uePXzeNDF4L3-_YAwyO2KQgcUKOrybr5ttwtaSuwo/edit?usp=sharing

The exact steps I followed are

Clone the issue#5 branch
Added .env file
Run docker-compose up
Access the "http://localhost:8000/swagger/" endpoint

Reid-T-W commented:
> When accessing the (GET) /userauth/groups/ endpoint i get the following error, kindly take a look via this link
> 
> https://docs.google.com/document/d/1WjOPYxjvWwAC4TicvSGyfkg62Hy9F1oQos3Q4PGftn4/edit?usp=sharing
> 
> The exact steps I followed are
> 
> Clone the issue#5 branch Added .env file Run docker-compose up Access the (GET) /userauth/groups/ endpoint via swagger

As per our conversation I tried to manually export the env variables, but the issue persisted. But it seems like manually running the python3 manage.py migrate command in the "cubeseed-backend-dev" container solves the issue.

sgarciahelguera commented:
> > When accessing the (GET) /userauth/groups/ endpoint i get the following error, kindly take a look via this link
> > https://docs.google.com/document/d/1WjOPYxjvWwAC4TicvSGyfkg62Hy9F1oQos3Q4PGftn4/edit?usp=sharing
> > The exact steps I followed are
> > Clone the issue#5 branch Added .env file Run docker-compose up Access the (GET) /userauth/groups/ endpoint via swagger
> 
> As per our conversation, I tried to export the env variables manually, but the issue persisted. However, the issue disappeared when the "python3 manage.py migrate" was run in the "cubeseed-backend-dev" container manually.

migration is run as a precondition, it should be done every time you update 
Maybe you had some changes come up and didn't update the container? 
Next time you see something like it try to do `docker-compose -f docker/docker-compose-dev.yml up --build`  (or with the corresponding yml file), and if it still fails try adding a verbose flag, it may add some useful info (or just a lot of noise) : `docker-compose --verbose -f docker/docker-compose.yml up --build`

sgarciahelguera committed:
Remove gunicorn, need to solve static files for it to work yet.

sgarciahelguera commented:
I just removed gunicorn, went back to django's runserver. I need to solve static files for gunicorn to work properly, I will do that for production together with testing AWS cloud settings.

I think I mentioned it before, but the entrypoint for the docker container is now a script that loads environment from a `.env` file if present, so that should solve some previous issues and simplify things.

Reid-T-W commented:
> > > When accessing the (GET) /userauth/groups/ endpoint i get the following error, kindly take a look via this link
> > > https://docs.google.com/document/d/1WjOPYxjvWwAC4TicvSGyfkg62Hy9F1oQos3Q4PGftn4/edit?usp=sharing
> > > The exact steps I followed are
> > > Clone the issue#5 branch Added .env file Run docker-compose up Access the (GET) /userauth/groups/ endpoint via swagger
> > 
> > 
> > As per our conversation, I tried to export the env variables manually, but the issue persisted. However, the issue disappeared when the "python3 manage.py migrate" was run in the "cubeseed-backend-dev" container manually.
> 
> migration is run as a precondition, it should be done every time you update Maybe you had some changes come up and didn't update the container? Next time you see something like it try to do `docker-compose -f docker/docker-compose.yml up --build` (or with the corresponding yml file), and if it still fails try adding a verbose flag, it may add some useful info (or just a lot of noise) : `docker-compose --verbose -f docker/docker-compose.yml up --build`

I didn't make any changes, but to be on the safe side I recloned the repo (branch issue#5) and run the command you suggested `docker-compose --verbose -f docker/docker-compose.yml up --build` but the issue persists. I understand that migration is being run as a precondition in a separate container, but it seems like the backend container is not accessing those migrations, can the issue be with the volume these containers are sharing?

sgarciahelguera commented:
OK, I was trying `docker/docker-compose.yml` and suspect you were trying the one in the root.
The issue is exactly what you say, but not because of a shared volume, sqlite lives on the local filesystem of the container.
So the migrations are applied in a container, not in the other.

Reid-T-W commented:
> OK, I was trying `docker/docker-compose.yml` and suspect you were trying the one in the root. The issue is exactly what you say, but not because of a shared volume, sqlite lives on the local filesystem of the container. So the migrations are applied in a container, not in the other.

Okay, that's a valid point, I think we can raise this as a separate issue and merge the changes, should we do that?

sgarciahelguera committed:
Move migrate into runner script, remove migration on all docker
compose files.

sgarciahelguera commented:
I moved the migrate command into the runner script (`docker/runner.sh`) and removed migration from docker compose files.
I did a quick run for both `docker/docker-compose.yml` and `docker-compose.yml` and it looks like they are running.
docker compose from github built images is not tested yet, I'll leave it for later.

Reid-T-W commented:
Great, it works on my end as well.
